pkg/registry/populator.go

```diff
@@ -145,25 +132,34 @@ func (i *DirectoryPopulator) globalSanityCheck(imagesToAdd []*ImageInput) error
 	return utilerrors.NewAggregate(errs)
 }
 
-func (i *DirectoryPopulator) loadManifests(imagesToAdd []*ImageInput, imagesToReAdd []*ImageInput, mode Mode) error {
+func (i *DirectoryPopulator) loadManifests(imagesToAdd []*ImageInput, mode Mode) error {
 	// global sanity checks before insertion
 	if err := i.globalSanityCheck(imagesToAdd); err != nil {
 		return err
 	}
 
 	switch mode {
 	case ReplacesMode:
-		for pkg := range i.overwriteDirMap {
-			// TODO: If this succeeds but the add fails there will be a disconnect between
-			// the registry and the index. Loading the bundles in a single transactions as
-			// described above would allow us to do the removable in that same transaction
-			// and ensure that rollback is possible.
-			if err := i.loader.RemovePackage(pkg); err != nil {
-				return err
+		// TODO: If this succeeds but the add fails there will be a disconnect between
```

Comment:
re this todo:

I think we also already don't really have transactional safety, so this shouldn't be a dealbreaker. most users will pull out the db from an image anyway, so that should be trivial to retry. imo this is something we should avoid implementing given the complexity of threading the transaction through without the right abstractions already in place

Reply:
agreed -- I think we should drop this comment